### ISSUE TITLE
Can't commit on arch linux (code path is wrong)

### ISSUE THREAD

**Issue by aksell** [kahole/edamagit#73]
Hey, this tool looks great! And all but one thing seems to work for me.

I'm having an issue when trying to commit. The editor window when attempting to commit is closed instantly, and there is an error from git:
```shell
"/usr/lib/electron7/bin/code" --wait: /usr/lib/electron7/bin/code: No such file or directory
error: There was a problem with the editor '"/usr/lib/electron7/bin/code" --wait'.
```
I'm using arch linux, and my code executable is at `/usr/bin/code`. Is there some way for me to manually specify where my code executable is at?

Thanks for writing this :raised_hands:

**Comment by kahole** [comment]
**TL;DR:** Running globally installed Electron makes Edamagit unable to locate VSCode start script.


Edamagit used to assume it was always in the users' path, but I moved away from that in order to make it easier to get started. (Especially on mac where it's usually not in the path by default)

Anyways, the `code` entity Edamagit is trying to reference here is just a bash-script for starting Electron/VSCode included by VSCode for convenience.

Is your `/usr/bin/code` the directory containing your VSCode installation? Or is it a symlink to the script located somewhere else?

The solution I went for is supposed to locate the actual location of this script, within the VSCode installation folder, based on the path of the currently running executable. (On Linux the `code`-script is usually located in the bin folder of the VSCode-installation itself)

The thing that's different with your setup is that you're running a globally installed Electron, which means the path of the current executable doesn't coincide with your VSCode installation.

**Some ways to solve:**
- Edamagit should try and detect this somehow and fall back to the users `path`.
- If not possible: Recommended people run VSCode with its included Electron to avoid problems

Thank you for reporting this, it didn't really occur to me that someone could run VSCode from outside its installation folder.
Solving this will make Edamagit more robust.

**Comment by kahole** [comment]
v0.4.3 is published on marketplace now.
It includes a fix that checks for the existence of the `code`-script at the process path location. If it can't find it, it falls back on the user path.
Let me know how it works out.

**Comment by aksell** [comment]
Thanks for the quick reply!
Your question: `/usr/bin/code` is symlinked to `/usr/bin/code-oss` (i'm not running VSCode, but Code - OSS, I don't know if that makes a difference). 
The contents of `/usr/bin/code-oss` are:
```bash
#!/bin/bash

ELECTRON_RUN_AS_NODE=1 exec electron7 /usr/lib/code/out/cli.js /usr/lib/code/code.js "$@"
```

**Comment by aksell** [comment]
With the update I can commit, all is great :+1:
